layout.ld

```diff
@@ -51,8 +50,8 @@ SECTIONS {
         LONG(_got);
         /* Size of GOT section */
         LONG(SIZEOF(.got));
-        /* Offset of data symbols in flash */
-        LONG(LOADADDR(.data) - _beginning);
+        /* Address of data symbols in flash */
+        LONG(_etext);
```

Comment:
This has changed from an offset to an address. An address is fine for static linking, but it will break when we have dynamically loaded processes (e.g., with relocation). I'd suggest changing this back to an offset, to match libtock-c. I think this line should not be needed if `.data` is being specified correctly.

Reply:
Pushed new revision. Please review. The requested change needed a following additional change.

```patch
+    let data_flash_start_addr = app_start + layout_header.data_sym_start;
+
     intrinsics::copy_nonoverlapping(
-        (text_start + layout_header.data_sym_start) as *const u8,
-        stack_top as *mut u8,
+        data_flash_start_addr as *const u8,
+        stacktop as *mut u8,
         layout_header.data_size,
     );
```